### PR TITLE
Shift Logger Complete!

### PR DESCRIPTION
**Pull request by Johay90**

### CODE REVIEW

File: Johay90.ShiftLogger/ShiftLoggerUI/UI/UserInputManager.cs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+        return employee;
+    }
+
+    private static (DateTime StartTime, DateTime EndTime) CollectShiftInfo()
+    {
+        var StartTime = AnsiConsole.Ask<DateTime>("Start Time? (mm-dd-yyyy HH:mm:ss)");
+        var EndTime = AnsiConsole.Ask<DateTime>("End Time? (mm-dd-yyyy HH:mm:ss)");
+
+        return (StartTime, EndTime);
```
Comment (javedkhan2k2) [inline Johay90.ShiftLogger/ShiftLoggerUI/UI/UserInputManager.cs]: StartTime and EndTime should be validated before sending back.

**Review by javedkhan2k2 — CHANGES_REQUESTED** [review]
@Johay90 good work.
The application is working without errors. You used http://NSwag.org to use as restclient. Which generated very ugly and unnecessary code. I will suggest that you create your own UI client service to handle all the api calls. This is only suggestion.
The only problem is that the shift start and end date is not validated in UI.
Kindly fix this issue.
You also mixed userInput and Displaying the content to user in UserinputManager.cs file. Try to use separate classes to implement each. Create a Visualisation class to handle all the outputing.

**Comment by Johay90** [comment]
Thanks for the review @javedkhan2k2 

I finally found time to implement the changes you recommended today. Summary of the changes:

1. Custom API Client: I've replaced the NSwag-generated client with a custom APIClient class.
2. Shift Validation in UI: I've added validation for shift start and end times directly in the UI.
3. Created a new Visualization class to handle all output operations

I ran a few brief tests and all seem to be working normally.

File: Johay90.ShiftLogger/ShiftLoggerUI/UI/UserInputManager.cs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+        private static (DateTime StartTime, DateTime EndTime) CollectShiftInfo()
+        {
+            DateTime startTime, endTime;
+            TimeSpan shiftLength;
+            bool isValid;
+
+            do
+            {
+                startTime = AnsiConsole.Ask<DateTime>("Start Time? (mm-dd-yyyy HH:mm:ss)");
```
Comment (javedkhan2k2) [inline Johay90.ShiftLogger/ShiftLoggerUI/UI/UserInputManager.cs]: The datetime is validating, but there is a small bug.
Now the user can enter dates without time, therefore default value is entered to fill the time part.
But in this application we need to expect the user to enter date and time.
Kindly create a custom validation to insure that the user entered the both date and time.

**Review by javedkhan2k2 — CHANGES_REQUESTED** [review]
@Johay90 
Now the client API is more readable and you are in control.
kindly fix the small bug and commit back.

File: Johay90.ShiftLogger/ShiftLoggerUI/UI/UserInputManager.cs (hunk 2 above)
Comment (Johay90) [reply Johay90.ShiftLogger/ShiftLoggerUI/UI/UserInputManager.cs]: Pushed, now requires the exact "MM-dd-yyyy HH:mm:ss" format for shift times before processing. 

Thank you!

**Comment by javedkhan2k2** [comment]
@Johay90 Approved, and well done.
You can still improved the structure of the project.
Try to create different classes(validation, controller) to divide the application logic. 
Good luck with your next project.

**Pull request closed without merging**